.github/workflows/test_on_push.yml

```diff
@@ -109,144 +109,22 @@ jobs:
         if: matrix.os != 'windows-latest'
         run: python -m nox -s pybamm-requires
 
-      - name: Run unit tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
-        run: python -m nox -s unit
-
-  check_coverage:
-    needs: style
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    name: Coverage tests (ubuntu-latest / Python 3.12)
-
-    steps:
-      - name: Check out PyBaMM repository
-        uses: actions/checkout@v4
-
-      - name: Install Linux system dependencies
-        uses: awalsh128/cache-apt-pkgs-action@v1.4.2
-        with:
-          packages: gfortran gcc graphviz pandoc
-          execute_install_scripts: true
-
-      # dot -c is for registering graphviz fonts and plugins
-      - name: Install OpenBLAS and TeXLive for Linux
+      - name: Run unit/coverage and integration tests
+        shell: bash
         run: |
-          sudo apt-get update
-          sudo dot -c
-          sudo apt-get install libopenblas-dev texlive-latex-extra dvipng
-
-      - name: Set up Python 3.12
-        id: setup-python
-        uses: actions/setup-python@v5
-        with:
-          python-version: 3.12
-          cache: 'pip'
-
-      - name: Install nox
-        run: python -m pip install nox
-
-      - name: Cache pybamm-requires nox environment for GNU/Linux
-        uses: actions/cache@v4
-        with:
-          path: |
-            # Repository files
-            ${{ github.workspace }}/pybind11/
-            ${{ github.workspace }}/install_KLU_Sundials/
-            # Headers and dynamic library files for SuiteSparse and SUNDIALS
-            ${{ env.HOME }}/.local/lib/
-            ${{ env.HOME }}/.local/include/
-          key: nox-${{ matrix.os }}-pybamm-requires-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/install_KLU_Sundials.py', '**/noxfile.py', '**/test_on_push.yml') }}
-
-      - name: Install SuiteSparse and SUNDIALS on GNU/Linux
-        timeout-minutes: 10
-        run: python -m nox -s pybamm-requires
-
-      - name: Run unit tests for Ubuntu with Python 3.11 and generate coverage report
-        run: python -m nox -s coverage
+          if [ ${{ matrix.python-version }} == "3.12" && ${{ matrix.os }} == "ubuntu-latest" ]; then
+            python -m nox -s coverage
+          else
+            python -m nox -s unit
+          fi
+          python -m nox -s integration
 
       - name: Upload coverage report
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.12'
         uses: codecov/codecov-action@v4.4.1
```

Comment:
This isn't running in any jobs right now because the Linux 3.12 job gets excluded. You shall have to remove the `exclude:` item from the matrix defined above